source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -4560,6 +4600,16 @@ export class DateInput extends Input {
             this._dateInputElement.value = this.defaultValue;
         }
 
+        if (this.inputStyle !== null && this.inputStyle === Enums.InputStyle.ReadWrite) {
+            this._dateInputElement.classList.add(this.hostConfig.makeCssClassName("ac-dateInput-readWriteStyle"));
+            Utils.SetDateOrTimePickerVisibility(this._dateInputElement, this.defaultValue, false /*show*/);
+
+            const handleHoverOnCard = (input: HTMLInputElement, eventType: Enums.CardEventType) => 
+                Utils.SetDateOrTimePickerVisibility(input, this._dateInputElement.value, eventType === Enums.CardEventType.OnMouseEnterOnCard /*show*/);
+
+            this.registerEventsForReadWriteInputStyle( this._dateInputElement, handleHoverOnCard.bind(this));
```

Comment:
Can you please explain what this does? I'm not getting it nor why it would be necessary?